### PR TITLE
Update to Sphinx>=3.2.0 using C domain

### PR DESCRIPTION
**Pull request by utzig**
Depends on https://github.com/nrfconnect/sdk-mcuboot/pull/123 and https://github.com/nrfconnect/sdk-nrfxlib/pull/247

### CODE REVIEW

**Comment by utzig** [comment]
Prereqs for this:

1) zephyr PR must be merged
2) CI must be updated to use Sphinx>=3.2.0 and Breathe>=4.20.0
2) `sdk-zephyr` must be synchronized
3) `nrfxlib` PR must be merged

**Comment by utzig** [comment]
> Prereqs for this:
> 
> 1. zephyr PR must be merged
> 2. CI must be updated to use Sphinx>=3.2.0 and Breathe>=4.20.0
> 3. `sdk-zephyr` must be synchronized
> 4. `nrfxlib` PR must be merged

And all of them are under review; will remove DNM when everything looks OK!

**Review by pdunaj — APPROVED** [review]

File: include/bluetooth/mesh/gen_plvl_srv.rst
Comment (trond-snekvik) [inline include/bluetooth/mesh/gen_plvl_srv.rst:46]: Can enumerators drop the explicit linking now?

**Review by trond-snekvik — APPROVED** [review]

**Review by doki-nordic — APPROVED** [review]

**Comment by carlescufi** [comment]
@utzig is this ready to merge or do we need a new upmerge from upstream?

**Comment by utzig** [comment]
> @utzig is this ready to merge or do we need a new upmerge from upstream?

Not ready yet, `sdk-zephyr` has to be synchronized (if it is I've missed it!), the CI docker image has to be updated with the latest tooling, also I am waiting for a Sphinx PR to be reviewed/merged to fix a few of the remaining issues.

**Review by joerchan — APPROVED** [review]

**Comment by joerchan** [comment]
After rebasing a new round is probably needed. Since this one is affecting all samples documentation it would be a good idea to complete it as soon as possible.

**Comment by Andrewpini** [comment]
Note: [PR-2816](https://github.com/nrfconnect/sdk-nrf/pull/2816) will be using the new enumerator type in its documentation, so please consider these files as well.

**Comment by utzig** [comment]
> Note: [PR-2816](https://github.com/nrfconnect/sdk-nrf/pull/2816) will be using the new enumerator type in its documentation, so please consider these files as well.

If you use `:c:enumerator` it will not pass the current CI until this PR and associated PRs are merged. Also you should use only `:c:enumerator:`ENUMERATOR` instead of `:cpp:enumerator:`ENUMERATOR <some_namespace::ENUMERATOR>` like it was used before. But I think it would be more "agile" to use old style and get your PR merged.

File: include/bluetooth/mesh/gen_plvl_srv.rst
Comment (utzig) [reply include/bluetooth/mesh/gen_plvl_srv.rst]: Sorry, missed this; unfortunately enumerators are broken in intersphinx for Sphinx==3.2.1, no matter if using explicit liking or not, but it's already fixed in `3.2.x` branch so when 3.2.2 comes out it should work without explicit linking.

**Comment by utzig** [comment]
@carlescufi @tejlmand This PR should be OK for merging now. The process for merging consists in

1) Merge a PR I have for the documentation CI, to skip a known warning in Sphinx 3 (an issue that also exists in 2, but does not generate a warning!)
2) Test this PR with the Docker image that has Sphinx==3.2.1 + Breathe==4.22.1 (Thomas knows how to do it!)
3) Make the updated Docker image default for all builds
4) Merge https://github.com/nrfconnect/sdk-nrfxlib/pull/247 and https://github.com/nrfconnect/sdk-mcuboot/pull/123 (this PR has one commit which fixes an issue with `recommonmark==0.6.0`, but I am not sure I am following proper procedure for a sdk-mcuboot PR)
5) Update west.yml and merge this, yay!

Nothing can go wrong really, haha, but I'd rather be online when the process is executed! :-P

**Review by KAGA164 — APPROVED** [review]

**Review by MarekPorwisz — APPROVED** [review]

**Comment by thst-nordic** [comment]
I'll do steps 1,2,3 right now.

**Comment by thst-nordic** [comment]
I'll do steps 1,2,3 right now.

**Review by thst-nordic — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Comment by tejlmand** [comment]
Please update manifest:
mcuboot sha: 82828b5cee28499157c340454e8cfa7a9beecd98
nrfxlib sha: 996b87e0ac42c5fc1b0f34100129f7184899a888

**Review by tejlmand — APPROVED** [review]

**Pull request merged**
